30413/Grama Malina Bianca/Assignment 1/main.c

```diff
@@ -0,0 +1,214 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+typedef struct node {
+    int data;
+    struct node *next;
+} NodeT;
+
+NodeT *first, *last;
+
+void initializeSll() {
+    first = NULL;
+    last = NULL;
+}
+
+void addLast(int data) {
+    if (first == NULL) {
+        first = (NodeT *) malloc(sizeof(NodeT));
+        first->next = last;
+        first->data = data;
+        last = first;
+    } else {
+        NodeT *newElement = (NodeT *) malloc(sizeof(NodeT));
+        last->next = newElement;
+        newElement->next = NULL;
+        newElement->data = data;
+        last = newElement;
+    }
+}
+
+void addFirst(int data) {
+    if (first == NULL) {
+        first = (NodeT *) malloc(sizeof(NodeT));
+        first->next = last;
+        first->data = data;
+        last = first;
+    } else {
+        NodeT *newElement = (NodeT *) malloc(sizeof(NodeT));
+        newElement->data = data;
+        newElement->next = first;
+        first = newElement;
+    }
+}
+
+void removeElement(int data) {
+    NodeT *currentElement = (NodeT *) malloc(sizeof(NodeT));
+    *currentElement = *first;
```

Comment:
De ce e nevoie de alocarea asta și de referențierea asta? 🤔 
Nu mergea dacă scriai pur și simplu `NodeT * currentElement = first;` ?

Reply:
`6` e fix elementul care apare de mai multe ori în listă - vezi ce trebuie să modifici pentru a-l face să meargă

Reply:
Dar ar trebui să fie indiciu suficient să te prinzi

Reply:
pentru ca daca las NodeT * currentElement = first; imi da eroarea asta:
"Assignment_1(28989,0x1043f55c0) malloc: *** error for object 0x7f89f1e00010: pointer being freed was not allocated
Assignment_1(28989,0x1043f55c0) malloc: *** set a breakpoint in malloc_error_break to debug"
care nu prea inteleg ce vrea de la mine si nici nu stiu unde ar trebui sa pun breakpoint-ul mai exact